anta/models.py

```diff
@@ -505,13 +505,22 @@ def save_commands_data(self, eos_data: list[dict[str, Any] | str]) -> None:
             self.instance_commands[index].output = data
 
     def __init_subclass__(cls) -> None:
-        """Verify that the mandatory class attributes are defined."""
-        mandatory_attributes = ["name", "description", "categories", "commands"]
+        """Verify that the mandatory class attributes are defined and set name and description if not set."""
+        mandatory_attributes = ["categories", "commands"]
         for attr in mandatory_attributes:
             if not hasattr(cls, attr):
                 msg = f"Class {cls.__module__}.{cls.__name__} is missing required class attribute {attr}"
                 raise NotImplementedError(msg)
 
+        if not hasattr(cls, "name"):
+            cls.name = cls.__name__
+        if not hasattr(cls, "description"):
+            if not cls.__doc__:
+                # No doctsring - raise
+                msg = f"Cannot set the description for class {cls.name}, either set it in the class definition or add a docstring to the class."
+                raise AttributeError(msg)
+            cls.description = cls.__doc__.split(sep="\n", maxsplit=1)[0]
+
```

Comment:
When extracting the description from the docstring, should we also handle cases where the docstring might be empty or contain only whitespace?

Could do something like:
```
    if not hasattr(cls, "description"):
        doc = (cls.__doc__ or '').strip()
        if doc:
            cls.description = doc.split('\n', 1)[0]
        else:
            msg = (
                f"Cannot set the description for class {cls.name}. "
                f"Either set 'description' in the class definition or add a docstring to the class."
            )
            raise AttributeError(msg)
```